mono/utils/mono-threads-windows.c

```diff
@@ -193,6 +195,9 @@ mono_threads_suspend_begin_async_suspend (MonoThreadInfo *info, gboolean interru
 	CONTEXT context;
 	context.ContextFlags = CONTEXT_INTEGER | CONTEXT_CONTROL;
 	if (!GetThreadContext (handle, &context)) {
+		THREADS_SUSPEND_DEBUG ("SUSPEND FAILED (GetThreadContext), id=%p, err=%u\n", GUINT_TO_POINTER (id), GetLastError ());
+		result = ResumeThread (handle);
+		g_assert (result == 1);
```

Comment:
Could anyone else have suspended it? Is this over-asserting?

Reply:
We have had the same assert in case we fail to change state as well for a long time, so if they fire, some external code probably suspended our thread underneath us. If we suspend a thread using mono runtime (in one way or another) we already keep a suspend count on the thread, meaning that this method should only be called ones, so believe the assert could give us some valuable information if they fire.

Reply:
Sometimes I suspend threads in debugger to aid single threaded debugging. Granted, I do it rarely. People also suspend threads sometimes when applying Detours. I wonder if it would ever be useful vs. rarely misfire and take things down unnecessarily.